experiments/dataSources/SalesforceREST/EinsteinAnalytics/readme.md

```diff
@@ -0,0 +1,95 @@
+
+    - Add a leading `\` in front of any double quote in the query statement itself.  (Don’t replace the double quotes in the JSON wrapper you added in step 1. 
+
+    - Take out all line breaks.  The JSON block will be one long line.   It will look somthing like this: 
+
+``{"query":"q = load \"0Fb6g0000008aw2CAA/0Fc6g0000009ZKoCAM\";q = group q by (‘Close_Date_Year’, ‘Account_Type’);q = foreach q generate ‘Close_Date_Year’ as ‘Close_Date_Year’, ‘Account_Type’ as ‘Account_Type’, sum('Amount') as 'sum_Amount';q = order q by 'sum_Amount' desc;q = limit q 2000;"}``
+
+
+#### Create a new REST model on your Skuid page
+
+Properties: 
+- Data Source URL:  “/services/data/v42.0/wave/query”
+- HTTP Verb:  “POST” 
+- Request body contents “As Templated Request Body”
+- Template:  Copy the JSON block you created above. 
+
+- You will need to swap over to the XML and add a “Path to Contents” entry in the highest level node of the model.   Add this to the end of the node  ``<model id=” …    pathtocontent="results.records" ``
+
+- Skuid cannot retrieve the model metadata in the builder, so you will need to manually create the fields.  Copy each of the items in the “generate” line of the query payload.  So for example in the query above you need to add fields with ID  ‘Close_Date_Year’,  ‘Account_Type’ and 'sum_Amount’  
+
+Now you should have the records in a skuid page that you can use in a chart, in a table, or any other visual component. 
+
+
+#### A note about filtering
+
+You can add “filter by” statements into your query.  See below
+
+```
+q = load \"0Fb6g0000008aw2CAA/0Fc6g0000009ZKoCAM\";
+q = filter q by 'Close_Date_Year' == \"2016\";
+q = group q by ('Industry', 'Product_Family');
+q = foreach q generate 'Industry' as 'Industry', 'Product_Family' as 'Product_Family', sum('Amount') as 'sum_Amount';
+q = order q by 'sum_Amount' desc;
+q = limit q 2000;
+```
+
+To make these interactive - you might think you could firs add merge syntax to replace the hard coded year.  And then find that Merge in model Condition. __But no.__  Because this merge is in the POST Request Body Template - it is not discovered as a “URL Merge” item and available to conditions. (There is an improvement in the works for this). 
+
+But there is a workaround: 
+
+- Create a UI-Only model and picklist field. 
+- Use Global merge syntax to push that field value into the POST Request Body Template. Like this: 
+``q = filter q by 'Close_Date_Year' == \"{{$Model.UiOnly.data.0.year}}\";``
+- Add a field editor to your page with that UI-Only picklist field.  (the button group display type for picklist fields is great here). 
+- Add a model action to that UI-Only model such that when the Year field is updated - your Analytics model is requeried.  
+
+
+## A sample dashboard. 
+
+Create a new V2 Page and use the XML found in “Einstein_Analytics_DTC_Dashboard”  to see a simple dashboard that is filterable by year.  This dashboard is a combination of charts and KPI blocks.  
```

Comment:
Need to note the error message that we know people will see on the sample page. 

Github suggestions can't do multi-line edits, so I'd probably break off this note to a new line, but here's what I would suggest adding.
```suggestion
Create a new V2 Page and use the XML found in “Einstein_Analytics_DTC_Dashboard”  to see a simple dashboard that is filterable by year.  This dashboard is a combination of charts and KPI blocks.  **Note**: You will see an error similar to the following within the App Composer: `1: Error retrieving metadata for Model(s) associated with Data Source '...'. Please check Models' properties to ensure they are set correctly. Error connecting to REST Data Source at URL "https://<<myDomain>>.my.salesforce.com/services/data/v42.0/wave/query": Expected query or query name in body.` *This is expected*.  It's because Skuid doesn't use our custom template when fetching metadata, but only at build time. The runtime—when you preview the page—will work as intended.
```